File: language-analysis/2.1.Concepts-Implementation/E-AbstractDataTypesAndEncapsulation/bank_account_adt.py
class BankAccount:
    def __init__(self, owner, balance):
        self.owner = owner
        self._balance = balance

    def deposit(self, amount):
        self._balance += amount

    def withdraw(self, amount):
        if amount <= self._balance:
            self._balance -= amount

    def get_balance(self):
        return self._balance


account = BankAccount("Cedric", 1000)
account.deposit(500)
account.withdraw(200)
print(account.get_balance())

File: language-analysis/2.1.Concepts-Implementation/E-AbstractDataTypesAndEncapsulation/bank_account_adt.js
class BankAccount {
  constructor(owner, balance) {
    this.owner = owner;
    this._balance = balance;
  }

  deposit(amount) {
    this._balance += amount;
  }

  withdraw(amount) {
    if (amount <= this._balance) {
      this._balance -= amount;
    }
  }

  getBalance() {
    return this._balance;
  }
}

const account = new BankAccount("Cedric", 1000);
account.deposit(500);
account.withdraw(200);
console.log(account.getBalance());
